peer-monitoring-service/client/src/network.rs

```diff
@@ -63,6 +63,16 @@ impl<NetworkClient: NetworkClientInterface<PeerMonitoringServiceMessage>>
     pub fn get_peers_and_metadata(&self) -> Arc<PeersAndMetadata> {
         self.network_client.get_peers_and_metadata()
     }
+
+    /// Disconnect from peer
+    pub async fn disconnect_from_peer(
+        &self,
+        peer: PeerNetworkId,
+        reason: String,
+    ) -> Result<(), aptos_network::application::error::Error> {
+        // TODO revisit if tokio::spawn is possible here and update the trait definition
+        self.network_client.disconnect_from_peer(peer, reason).await
```

Comment:
Nice. Yeah, let's maybe also mark the connection state as disconnecting, too. Just to be consistent with the existing implementation: https://github.com/aptos-labs/aptos-core/blob/310f98bc87e73432c24f69d47536b323d7794866/network/framework/src/protocols/health_checker/interface.rs#L67

Reply:
I thought I did this already on line 79 @JoshLind? Maybe i'm misunderstanding...